### PR TITLE
add sysctl init container for pd (#3347)

### LINKED ISSUE #3241: Support pod SecurityContext for PD
## Feature Request

**Is your feature request related to a problem? Please describe:**

The pod SecurityContext has been supported for TiDB and TiKV in https://github.com/pingcap/tidb-operator/pull/1107, it's better to support for PD too.
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #3347 to release-1.1

---

### What problem does this PR solve? 
fix #3241

### What is changed and how does it work?
add sysctl init container for PD

### Check List 

Tests 

 - Unit test
 - E2E test

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by DanielZhangQD** [comment]
@shonge Please resolve the conflict.

File: pkg/manager/member/pd_member_manager.go [hunk 1]
```diff
@@ -720,7 +759,12 @@ func getNewPDSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (
 	if podSpec.ServiceAccountName == "" {
 		podSpec.ServiceAccountName = tc.Spec.ServiceAccount
 	}
+<<<<<<< HEAD
```
Comment (shonge) [inline pkg/manager/member/pd_member_manager.go:762]:
```suggestion
```

File: pkg/manager/member/pd_member_manager.go [hunk 2]
```diff
@@ -720,7 +759,12 @@ func getNewPDSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (
...
 
+=======
```
Comment (shonge) [inline pkg/manager/member/pd_member_manager.go:764]:
```suggestion
```

File: pkg/manager/member/pd_member_manager.go [hunk 3]
```diff
@@ -720,7 +759,12 @@ func getNewPDSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (
...
+	podSpec.SecurityContext = podSecurityContext
+	podSpec.InitContainers = initContainers
+>>>>>>> 866957b0... add sysctl init container for pd (#3347)
```
Comment (shonge) [inline pkg/manager/member/pd_member_manager.go:767]:
```suggestion
```

File: pkg/manager/member/pd_member_manager_test.go [hunk 4]
```diff
@@ -951,6 +951,11 @@ func testAdditionalVolumes(t *testing.T, additionalVolumes []corev1.Volume) func
 
 func TestGetNewPDSetForTidbCluster(t *testing.T) {
 	enable := true
+<<<<<<< HEAD
```
Comment (shonge) [inline pkg/manager/member/pd_member_manager_test.go:954]:
```suggestion
```

File: pkg/manager/member/pd_member_manager_test.go [hunk 5]
```diff
@@ -951,6 +951,11 @@ func testAdditionalVolumes(t *testing.T, additionalVolumes []corev1.Volume) func
...
+=======
```
Comment (shonge) [inline pkg/manager/member/pd_member_manager_test.go:955]:
```suggestion
```

File: pkg/manager/member/pd_member_manager_test.go [hunk 6]
```diff
@@ -951,6 +951,11 @@ func testAdditionalVolumes(t *testing.T, additionalVolumes []corev1.Volume) func
...
+	asNonRoot := true
+	privileged := true
+>>>>>>> 866957b0... add sysctl init container for pd (#3347)
```
Comment (shonge) [inline pkg/manager/member/pd_member_manager_test.go:958]:
```suggestion
```

File: pkg/manager/member/pd_member_manager_test.go [hunk 7]
```diff
@@ -1332,6 +1337,358 @@ func TestGetNewPDSetForTidbCluster(t *testing.T) {
 			},
 			testSts: testAdditionalVolumes(t, []corev1.Volume{{Name: "test", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}}}),
 		},
+<<<<<<< HEAD
```
Comment (shonge) [inline pkg/manager/member/pd_member_manager_test.go:1340]:
```suggestion
```

File: pkg/manager/member/pd_member_manager_test.go [hunk 8]
```diff
@@ -1332,6 +1337,358 @@ func TestGetNewPDSetForTidbCluster(t *testing.T) {
...
+=======
```
Comment (shonge) [inline pkg/manager/member/pd_member_manager_test.go:1341]:
```suggestion
```

File: pkg/manager/member/pd_member_manager_test.go [hunk 9]
```diff
@@ -1332,6 +1337,358 @@ func TestGetNewPDSetForTidbCluster(t *testing.T) {
...
+					TiKV: &v1alpha1.TiKVSpec{},
+				},
+			},
+			testSts: func(sts *apps.StatefulSet) {
+				g := NewGomegaWithT(t)
+				g.Expect(sts.Spec.Template.Spec.InitContainers).Should(BeEmpty())
+				g.Expect(sts.Spec.Template.Spec.SecurityContext).To(BeNil())
+			},
+		},
+>>>>>>> 866957b0... add sysctl init container for pd (#3347)
```
Comment (shonge) [inline pkg/manager/member/pd_member_manager_test.go:1691]:
```suggestion
```

**Comment by DanielZhangQD** [comment]
Close via #3384.

**Pull request closed without merging**